api/chartRepo/ChartRepositoryRestHandler.go

```diff
@@ -283,4 +305,89 @@ func (handler *ChartRepositoryRestHandlerImpl) DeleteChartRepo(w http.ResponseWr
 		return
 	}
 	common.WriteJsonResp(w, nil, CHART_REPO_DELETE_SUCCESS_RESP, http.StatusOK)
-}
\ No newline at end of file
+}
+
+func (handler *ChartRepositoryRestHandlerImpl) CreateChartFromBinary(w http.ResponseWriter, r *http.Request) {
+	userId, err := handler.userAuthService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, nil, http.StatusUnauthorized)
+		return
+	}
+
+	token := r.Header.Get("token")
+	if ok := handler.enforcer.Enforce(token, casbin.ResourceGlobal, casbin.ActionUpdate, "*"); !ok {
+		common.WriteJsonResp(w, errors.New("unauthorized"), nil, http.StatusForbidden)
+		return
+	}
+
+	var bean ChartBinary
+	file, fileHeader, err := r.FormFile("BinaryFile")
+	if err != nil {
+		handler.Logger.Errorw("request err, File parsing error", "err", err, "payload", file)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+
+	if err := r.ParseForm(); err != nil {
+		handler.Logger.Errorw("request err, Corrupted form data", "err", err, "payload", file)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+
+	bean.BinaryFileName = r.PostForm.Get("BinaryFileName")
+	bean.Version = r.PostForm.Get("Version")
+
+	fileBytes, err := ioutil.ReadAll(file)
```

Comment:
all code below this except response should be moved to service and not in resthandler

Reply:
extractTarGz should be reused

Reply:
can reuse part of CreateChartIfNotExists